scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -540,80 +626,82 @@ EnhancedVolcano::EnhancedVolcano(deseq_results,
+This can help us validate the `DESeq2` results so that we can visualize gene expression changes across our cell type of interest on a single-cell level. 
 
-# add back modified coldata containing gene expression for sig genes 
-colData(rms_sce) <- DataFrame(combined_coldata_df, row.names = combined_coldata_df$cell_id)
-```
-
-We will pick a gene and look at the expression of that gene across subdiagnosis, sample, and cell type.
-
-```{r}
+```{r expression umap}
 # filter to just myoblast cells and remove any NA's before plotting
 myoblast_combined_sce <- rms_sce[, which(rms_sce$celltype_broad == "Tumor_Myoblast")]
 
-# first look at just ARMS vs. ERMS 
+# plot PTPRT expression in ARMS vs. ERMS
 scater::plotReducedDim(myoblast_combined_sce,
                        dimred = "fastmnn_UMAP",
                        colour_by = "ENSG00000196090", #PTPRT
                        point_size= 0.5,
                        point_alpha = 0.4,
                        other_fields = "diagnosis_group") +
-  facet_wrap(~ diagnosis_group, nrow = 3)
+  facet_grid(cols = vars(diagnosis_group))
 ```
 
+In the above plot we only plotted the tumor myoblast cells that we used in our DE analysis. 
+However, we might be interested to see the expression of genes that are differentially expressed in other cell types present in our samples.
 
-```{r}
-# let's look across some other cell types
-celltypes <- c("Tumor_Myoblast", "Tumor_Mesoderm", "Tumor_Myocyte", "Vascular Endothelium")
+```{r celltype comparison}
+# let's compare gene expression across some other cell types
+# look at all tumor cells and pick one normal cell type
+celltypes <- c("Tumor_Myoblast", 
+               "Tumor_Mesoderm", 
+               "Tumor_Myocyte", 
+               "Vascular Endothelium")
 
 # subset to just tumor celltypes that we are interested in
 tumor_sce <- rms_sce[, which(rms_sce$celltype_broad %in% celltypes)]
 ```
 
+Next we will look at a few DE genes that we identified, one up regulated gene and one down regulated gene, and compare their expression in myoblasts to other cell types in ARMS and ERMS samples.
+We will use the `scater::plotExpression()` function to create a violin plot with RMS subtype on the x-axis and gene expression on the y-axis.
+We can continue using `facet_grid()` to show separate panels for each cell type.
+Because we want to show multiple genes here, we are going to add an additional option to `facet_grid()` to include multiple rows in our plot grid, one for each gene of interest.
 
-```{r}
-# pick a few genes to look at 
+```{r multi-gene plot}
+# pick a couple genes to look at 
 genes_to_plot <- c("ENSG00000196090", #PTPRT
                    "ENSG00000148935") #GAS2
 
-# create a violin plot across cell types of interest
+# create a violin plot 
 scater::plotExpression(tumor_sce,
-                       features = genes_to_plot, #PTPRT
+                       features = genes_to_plot, # a vector of genes to plot
```

Comment:
This may run into spacing issues. If so, 
```suggestion
                       # a vector of genes to plot
                       features = genes_to_plot, 
```